nntrainer/include/layers.h

```diff
@@ -144,12 +144,11 @@ class Layer {
 
   /**
    * @brief     set Property of layer
-   * @param[in] key key of property
    * @param[in] values values of property
    * @retval #ML_ERROR_NONE Successful.
    * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
    */
-  virtual int setProperty(const char *key, std::vector<std::string> values) = 0;
+  virtual int setProperty(std::vector<std::string> values) = 0;
```

Comment:
Recommendation:
- Layer::setProperty becomes a non-virtual concrete function that loops the values vector
- Create protected virtual functions setPropertyItem (int-key, value) and parseLayerProperty(str-key), where their default Layer class methods do "return NO_SUCH_KEY"
- Let each class worry about each key-value pair only.

Reply:
right. I will update this next pr. Thanks.